src/System.Collections.Immutable/src/System/Collections/Immutable/SortedInt32KeyNode.cs

```diff
@@ -200,8 +200,28 @@ internal SortedInt32KeyNode<TValue> Remove(int key, out bool mutated)
         [Pure]
         internal TValue GetValueOrDefault(int key)
         {
-            var match = this.Search(key);
-            return match.IsEmpty ? default(TValue) : match._value;
+            SortedInt32KeyNode<TValue> node = this;
+            while (true)
```

Comment:
Is it not worth sharing this and the below implementation?

Reply:
We could.  This helps a bit as it avoids an extra call and extra decision based on whether the node is empty, which would already have been made in the shared helper.  And it really only ends up being a few lines of shared code.  But I can change it if you'd prefer...?

Reply:
Simple enough that I think it's fine. =